command/root.go

```diff
@@ -109,11 +110,18 @@ func BasicClient() (*api.Client, error) {
 		opts = append(opts, apiVerboseLog())
 	}
 	opts = append(opts, api.AddHeader("User-Agent", fmt.Sprintf("GitHub CLI %s", Version)))
-	if c, err := context.ParseDefaultConfig(); err == nil {
-		if token, err := c.Get(defaultHostname, "oauth_token"); err == nil {
-			opts = append(opts, api.AddHeader("Authorization", fmt.Sprintf("token %s", token)))
-		}
+
+	c, err := config.ParseDefaultConfig()
+	if err != nil {
+		return nil, err
 	}
+
+	token, err := c.Get(defaultHostname, "oauth_token")
+	if token == "" || err != nil {
+		return nil, err
```

Comment:
If token was an empty string but there was no error object, this will return `nil, nil`. If we want to return an explicit error for that case, we should describe it to ensure there is always an error object:

```go
if err != nil {
  return nil, err
}
if token == "" {
  return nil, fmt.Errorf("...")
}
```

Reply:
good catch; i was going too fast